config/enabled.yml

```diff
@@ -1403,6 +1403,10 @@ Lint/UnreachableCode:
   Description: 'Unreachable code.'
   Enabled: true
 
+Lint/UriRegexp:
+  Description: 'Use URI::Parser.new.make_regexp` instead of `URI.regexp`.'
```

Comment:
I guess `URI::DEFAULT_PARSER.make_regexp` is better.
https://github.com/ruby/ruby/blob/4b9aeef1fa2f32a4555f88c6e83539374a1bd8d1/lib/uri/common.rb#L22


```ruby
require 'benchmark'
require 'uri'

Benchmark.bm(20) do |x|
  puts 'empty arg'
  x.report('URI::Parser.new'){    2000.times{URI::Parser.new.make_regexp}}
  x.report('URI::DEFAULT_PARSER'){2000.times{URI::DEFAULT_PARSER.make_regexp}}

  puts 'with arg'
  x.report('URI::Parser.new'){    2000.times{URI::Parser.new.make_regexp('http://example.com')}}
  x.report('URI::DEFAULT_PARSER'){2000.times{URI::DEFAULT_PARSER.make_regexp('http://example.com')}}
end
```

```
                           user     system      total        real
empty arg
URI::Parser.new        2.010000   0.010000   2.020000 (  2.016804)
URI::DEFAULT_PARSER    0.000000   0.000000   0.000000 (  0.000144)
with arg
URI::Parser.new        2.460000   0.000000   2.460000 (  2.451649)
URI::DEFAULT_PARSER    0.330000   0.000000   0.330000 (  0.331841)
```

Reply:
I see! Because it does not generate every time like Parser.new, the constant DEFAULT_PARSER is fast. I fixed by d6df59d.